scanner/matcher/matcher.go

```diff
@@ -26,20 +39,22 @@ import (
 )
 
 // matcherNames specifies the ClairCore matchers to use.
+//
+// Note: Do NOT hardcode the names. It's very easy to mess up...
 var matcherNames = []string{
-	"alpine-matcher",
-	"aws-matcher",
-	"debian-matcher",
-	"gobin",
-	"java-maven",
-	"oracle",
-	"photon",
-	"python",
-	"rhel-container-matcher",
-	"rhel",
-	"ruby-gem",
-	"suse",
-	"ubuntu-matcher",
+	(*alpine.Matcher)(nil).Name(),
+	(*aws.Matcher)(nil).Name(),
+	(*debian.Matcher)(nil).Name(),
+	(*gobin.Matcher)(nil).Name(),
+	(*java.Matcher)(nil).Name(),
+	(*oracle.Matcher)(nil).Name(),
+	(*photon.Matcher)(nil).Name(),
+	(*python.Matcher)(nil).Name(),
+	rhcc.Matcher.Name(),
+	(*rhel.Matcher)(nil).Name(),
+	(*ruby.Matcher)(nil).Name(),
+	(*suse.Matcher)(nil).Name(),
+	(*ubuntu.Matcher)(nil).Name(),
 }
```

Comment:
yeah it is, though perhaps it's more space efficient since it's `nil`? See the comment I added right above this for the reason why I did this. When I was first testing VEX support, I wasn't getting results because I messed up the name. I did this to future-proof things in case names change.

Note: I don't do this for the updaters in `export.go` below because those names aren't always exposed like these are